pkg/console/subresource/consoleserver/config_builder.go

```diff
@@ -317,32 +317,74 @@ func (b *ConsoleServerCLIConfigBuilder) customization() Customization {
 		conf.CustomLogoFile = b.customLogoFile
 	}
 
-	if b.devCatalogCustomization.Categories != nil {
-		mapMeta := func(meta operatorv1.DeveloperConsoleCatalogCategoryMeta) DeveloperConsoleCatalogCategoryMeta {
-			return DeveloperConsoleCatalogCategoryMeta{
-				ID:    meta.ID,
-				Label: meta.Label,
-				Tags:  meta.Tags,
+	if (!reflect.DeepEqual(b.devCatalogCustomization, operatorv1.DeveloperConsoleCatalogCustomization{})) {
+
+		var catalogTypes DeveloperConsoleCatalogTypesState
+		// var categories []DeveloperConsoleCatalogCategory
+		categories := make([]DeveloperConsoleCatalogCategory, len(b.devCatalogCustomization.Categories))
+
+		if b.devCatalogCustomization.Categories != nil {
+			mapMeta := func(meta operatorv1.DeveloperConsoleCatalogCategoryMeta) DeveloperConsoleCatalogCategoryMeta {
+				return DeveloperConsoleCatalogCategoryMeta{
+					ID:    meta.ID,
+					Label: meta.Label,
+					Tags:  meta.Tags,
+				}
+			}
+
+			// categories := make([]DeveloperConsoleCatalogCategory, len(b.devCatalogCustomization.Categories))
+			for categoryIndex, category := range b.devCatalogCustomization.Categories {
+				var subcategories []DeveloperConsoleCatalogCategoryMeta = nil
+				if category.Subcategories != nil {
+					subcategories = make([]DeveloperConsoleCatalogCategoryMeta, len(category.Subcategories))
+					for subcategoryIndex, subcategory := range category.Subcategories {
+						subcategories[subcategoryIndex] = mapMeta(subcategory)
+					}
+				}
+				categories[categoryIndex] = DeveloperConsoleCatalogCategory{
+					DeveloperConsoleCatalogCategoryMeta: mapMeta(category.DeveloperConsoleCatalogCategoryMeta),
+					Subcategories:                       subcategories,
+				}
 			}
 		}
+		if !reflect.DeepEqual(b.devCatalogCustomization.Types, operatorv1.DeveloperConsoleCatalogTypesState{}) {
 
-		categories := make([]DeveloperConsoleCatalogCategory, len(b.devCatalogCustomization.Categories))
-		for categoryIndex, category := range b.devCatalogCustomization.Categories {
-			var subcategories []DeveloperConsoleCatalogCategoryMeta = nil
-			if category.Subcategories != nil {
-				subcategories = make([]DeveloperConsoleCatalogCategoryMeta, len(category.Subcategories))
-				for subcategoryIndex, subcategory := range category.Subcategories {
-					subcategories[subcategoryIndex] = mapMeta(subcategory)
+			var disabledTypes []string = nil
+			var enabledTypes []string = nil
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeDisabled) && b.devCatalogCustomization.Types.Disabled != nil {
+				disabledTypes = *b.devCatalogCustomization.Types.Disabled
+			}
+
+			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypesState(CatalogTypeEnabled) && b.devCatalogCustomization.Types.Enabled != nil {
+				enabledTypes = *b.devCatalogCustomization.Types.Enabled
+			}
```

Comment:
The "enum" is just a typed string, comparing this with the local CatalogTypeDisabled needs a cast, but it might be easier to use the const from operatorv1:

```suggestion
			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypeDisabled && b.devCatalogCustomization.Types.Disabled != nil {
				disabledTypes = *b.devCatalogCustomization.Types.Disabled
			}

			if b.devCatalogCustomization.Types.State == operatorv1.CatalogTypeEnabled && b.devCatalogCustomization.Types.Enabled != nil {
				enabledTypes = *b.devCatalogCustomization.Types.Enabled
			}
```

There are some more spots for CatalogTypeDisabled and CatalogTypeEnabled

Reply:
But one more note here. Isn't it enough to copy the complete `catalogTypes` like this?

```go
		if !reflect.DeepEqual(b.devCatalogCustomization.Types, operatorv1.DeveloperConsoleCatalogTypesState{}) {
			catalogTypes = DeveloperConsoleCatalogTypesState{
				State:    CatalogTypesState(b.devCatalogCustomization.Types.State),
				Enabled:  b.devCatalogCustomization.Types.Enabled,
				Disabled: b.devCatalogCustomization.Types.Disabled,
			}
		}
```